webrender/src/device/gfx.rs

```diff
@@ -495,31 +496,22 @@ impl<B: hal::Backend> ImageCore<B> {
         .expect("create_image failed");
         let requirements = unsafe { device.get_image_requirements(&image) };
 
-        let mem_type = memory_types
-            .iter()
-            .enumerate()
-            .position(|(id, mem_type)| {
-                requirements.type_mask & (1 << id) != 0
-                    && mem_type
-                        .properties
-                        .contains(hal::memory::Properties::DEVICE_LOCAL)
-            })
-            .unwrap()
-            .into();
+        let memory_block = heaps.allocate(
+            device,
+            requirements.type_mask as u32,
+            MemoryUsageValue::Data,
+            requirements.size,
```

Comment:
I wonder why it simply don't re-export `Requirements` from gfx-hal that it's based on. cc @omni-viral

Reply:
No good. Reason. It could have `Requirements` argument instead.